### PR TITLE
Added functions for AWS by Aleksey63

### PR DESCRIPTION
**Pull request by gentee**

### CODE REVIEW

**Review by potehinre — APPROVED** [review]

File: packages/api/vde_test.go [hunk 1]
```diff
@@ -316,6 +316,34 @@ func TestVDEParams(t *testing.T) {
...
+	value := `{"en": "My VDE test", "fr": "French VDE test"}`
+
+	form = url.Values{"Name": {name}, "Trans": {value}, "vde": {`true`}}
+	err = postTx(`NewLang`, &form)
+	if err != nil {
+		t.Error(err)
+		return
+	}
+	input := fmt.Sprintf(`Span($%s$)+LangRes(%[1]s,fr)`, name)
+	var lret contentResult
```
Comment (ghost) [inline packages/api/vde_test.go]: please, use semantic names

File: packages/smart/funcs.go [hunk 2]
```diff
@@ -876,6 +907,50 @@ func IDToAddress(id int64) (out string) {
...
+	}
+	if raw_output {
+		return string(hash), nil
+	} else {
+		return hex.EncodeToString(hash), nil
+	}
+}
+
+//Returns the array of keys of the map
+func GetMapKeys(in map[string]interface{}) []interface{} {
```
Comment (ghost) [inline packages/smart/funcs.go:922]: useless comment

File: packages/smart/funcs.go [hunk 3]
```diff
@@ -876,6 +907,50 @@ func IDToAddress(id int64) (out string) {
...
+	keys := make([]interface{}, 0, len(in))
+	for k := range in {
+		keys = append(keys, k)
+	}
+	return keys
+}
+
+//Returns the sorted array of keys of the map
+func SortedKeys(m map[string]interface{}) []interface{} {
+	i, sorted := 0, make([]string, len(m))
```
Comment (ghost) [inline packages/smart/funcs.go:932]: useless comment

File: packages/smart/smart.go [hunk 4]
```diff
@@ -85,7 +85,11 @@ func GetTestValue(name string) string {
 
 // GetLogger is returning logger
 func (sc SmartContract) GetLogger() *log.Entry {
-	return log.WithFields(log.Fields{"vde": sc.VDE, "name": sc.TxContract.Name})
+	var name string
```
Comment (ghost) [inline packages/smart/smart.go:88]: useless comment

File: packages/smart/funcs.go [hunk 5]
```diff
@@ -916,3 +991,35 @@ func HTTPRequest(requrl, method string, headers map[string]interface{},
 	}
 	return string(data), nil
 }
+
+// HTTPPostJSON sends post http request with json
+func HTTPPostJSON(requrl string, headers map[string]interface{}, json_str string) (string, error) {
+
```
Comment (ghost) [inline packages/smart/funcs.go:995]: useless comment

**Review by ghost — CHANGES_REQUESTED** [review]

File: packages/api/vde_test.go (hunk 1 above)
Comment (gentee) [reply packages/api/vde_test.go]: It has been renamed to retContent.

**Review by ghost — APPROVED** [review]

**Pull request merged**